test-app/build-tools/android-metadata-generator/src/src/com/telerik/metadata/bcl/ClassInfo.java

```diff
@@ -63,6 +60,19 @@ private void init() {
         }
     }
 
+    private String getClassName(int classIndex) {
```

Comment:
It might be better not to return null if name is not resolved but java.util.Optional instead. Not an issue though.
The ConstantPool instance might be provided to the method as an argument and I think it's also better to make the method 'private static' as it is not modifying the state of the ClassInfo instance and is more like a helper method.

Reply:
Using Optional currently is a problem as the metadata generator project is inside an android project and thus Optional is not allowed as it requires minimum API level 24. About the static method, I prefer this way for simplicity, as you don't need to pass the constant pool additionally and being an instance method is should be intuitive that it uses the current class constant pool.